tests/apps/bookstore/bookstore-mp/src/main/java/io/helidon/tests/apps/bookstore/mp/BookResource.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2018, 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.tests.apps.bookstore.mp;
+
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Collection;
+import java.util.Optional;
+
+import io.helidon.tests.apps.bookstore.mp.Book;
+import io.helidon.tests.apps.bookstore.mp.BookStore;
+
+@Path("/books")
+@RequestScoped
+public class BookResource {
+
+    private final BookStore bookStore;
+
+    @Inject
+    public BookResource(BookStore bookStore) {
+        this.bookStore = bookStore;
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getBooks() {
+        Collection<Book> books = bookStore.getAll();
+        return Response.ok(books).build();
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response postBook(Book book) {
+        Optional<Book> optional = bookStore.find(book.getIsbn());
+        if (optional.isPresent()) {
+            return Response.status(Response.Status.CONFLICT).build();
+        }
+        bookStore.store(book);
+        return Response.ok().build();
+    }
+
+    @GET
+    @Path("{isbn}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getBook(@PathParam("isbn") String isbn) {
+        Optional<Book> optional = bookStore.find(isbn);
+        return optional.isPresent() ? Response.ok(optional.get()).build()
+                : Response.status(Response.Status.NOT_FOUND).build();
+
+    }
+
+    @PUT
+    @Path("{isbn}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response putBook(Book book) {
+        Optional<Book> optional = bookStore.find(book.getIsbn());
+        if (!optional.isPresent()) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+        bookStore.store(book);
+        return Response.ok().build();
```

Comment:
```
return bookStore.find(book.getIsbn())
                .map(found -> {
                    bookStore.store(book);
                    return Response.ok().build();
                })
                .orElseGet(() -> Response.status(Response.Status.NOT_FOUND).build());
```

Reply:
I don't agree that this is necessarily more readable; it is a matter of style. Also using lambdas to avoid flow control statements does result in _more objects being created_ (a problem that is apparent in our profiles!). In the example above, in particular, the second lambda has an implicit reference to "this" which would require a new instance of it on every call to the resource method.

Reply:
Having the lambda to `map` do more than just transform the Optional value feels wrong to me.

Reply:
Optional is now gone.